tools/cldr-code/src/main/java/org/unicode/cldr/util/CLDRFileOverride.java

```diff
@@ -0,0 +1,168 @@
+package org.unicode.cldr.util;
+
+import com.ibm.icu.util.VersionInfo;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Set;
+import org.unicode.cldr.util.XPathParts.Comments;
+
+public class CLDRFileOverride extends CLDRFile {
+
+    /**
+     * Overrides the values of the sourceFile. The keys of the overrides must already be in the
+     * XMLSource source
+     */
+    public CLDRFileOverride(CLDRFile sourceFile, Map<String, String> valueOverrides) {
+        super(new XMLSourceMapOverride(sourceFile.dataSource, valueOverrides));
+    }
+
+    /**
+     * Overrides the values of the source. <br>
+     * The keys of the overrides must already be in the XMLSource source as DPaths. And the full
+     * paths cannot be changed <br>
+     * Note: this is similar to the top-level DelegateXMLSource in the apps directory, but not the
+     * same, since it provides for the map.
+     */
+    public static class XMLSourceMapOverride extends XMLSource {
+        private XMLSource delegate;
```

Comment:
perhaps DelegateXMLSource should be modified so that this could be a subclass of it instead of largely duplicate?